collector/internal/telemetryapi/client.go

```diff
@@ -49,7 +49,7 @@ func NewClient(logger *zap.Logger) *Client {
 func (c *Client) Subscribe(ctx context.Context, extensionID string, listenerURI string) (string, error) {
 	eventTypes := []EventType{
 		Platform,
-		// Function,
+		Function,
```

Comment:
I feel like the choice of which data types (traces, logs) to process and emit should be driven by config.  But that might be out of scope for this PR and better addressed separately, since it would involve even more refactoring to limit the subscription to what's required, etc.